### ISSUE TITLE
Type ids vary by install

### ISSUE THREAD

**Issue by turnage** [reddit/reddit#1456]
On my local install of reddit, post types are t5, instead of t3. For the endpoint ````https://reddit.local/r/test.json````, the content is a listing of t5 types:

    {"kind": "Listing", "data": {"modhash": "", "children": [{"kind": "t5", "data": {"domain": "self.test", "banned_by": null, "media_embed": {}, "subreddit": "test", "selftext_html": null, "selftext": "", "likes": null, "suggested_sort": null, "user_reports": [], "secure_media": null, "link_flair_text": null, "id": "1", "gilded": 0, "archived": false, "clicked": false, "report_reasons": null, "author": "one", "media": null, "name": "t5_1", "score": 1, "approved_by": null, "over_18": false, "hidden": false, "thumbnail": "", "subreddit_id": "t4_1", "edited": false, "link_flair_css_class": null, "author_flair_css_class": null, "downs": 0, "mod_reports": [], "secure_media_embed": {}, "saved": false, "removal_reason": null, "is_self": true, "hide_score": false, "permalink": "/r/test/comments/1/test_post_please_ignore/", "locked": false, "stickied": false, "created": 1446686219.0, "url": "http://reddit.local/r/test/comments/1/test_post_please_ignore/", "author_flair_text": null, "quarantine": false, "title": "test post please ignore", "created_utc": 1446686219.0, "ups": 1, "num_comments": 0, "visited": false, "num_reports": null, "distinguished": null}}], "after": null, "before": null}}

````xiong_chiamiov```` from the reddit-dev irc suggested

> iirc, which Things have which ids isn't statically defined in code, so it can vary by install.

Can the type ids be configured so that they match the Reddit api? Are there any plans to make that the default?

**Comment by 13steinj** [comment]
Since this isn't an actual bug; you may want to post to /r/redditdev for a quicker response.

**Comment by turnage** [comment]
I can understand if this isn't a bug, but I do think it is an issue if isn't documented. Forgive me if I made a mistake and missed the instructions on it. [This is the closest I found](https://github.com/reddit/reddit/blob/c902b602933b0e02a6aa7f5364f517260617650c/scripts/migrate/udfs/src/com/reddit/pig/TypeID.java#L4)..

**Comment by spladug** [comment]
> Can the type ids be configured so that they match the Reddit api?

Yes, [type IDs can be statically defined](https://github.com/reddit/reddit/blob/master/r2/example.ini#L598-L601).  This config change would need to happen before the database is first initialized on the first run of the app; I've not tested doing such a thing, this feature was only added so that production (which obviously doesn't recreate the master databases often) could skip querying for typeids on startup.

To match production exactly:

```ini
db_table_link = thing, !typeid=3
db_table_account = thing, !typeid=2
db_table_message = thing, !typeid=4
db_table_comment = thing, !typeid=1
db_table_subreddit = thing, !typeid=5
db_table_srmember = relation, subreddit, account, !typeid=11
db_table_friend = relation, account, account, !typeid=9
db_table_inbox_account_comment = relation, account, comment, !typeid=1
db_table_inbox_account_message = relation, account, message, !typeid=2
db_table_moderatorinbox = relation, subreddit, message, !typeid=14
db_table_report_account_link = relation, account, link, !typeid=3
db_table_report_account_comment = relation, account, comment, !typeid=4
db_table_report_account_message = relation, account, message, !typeid=7
db_table_report_account_subreddit = relation, account, subreddit, !typeid=12
db_table_award = thing, !typeid=6
db_table_trophy = relation, account, award, !typeid=13
db_table_jury_account_link = relation, account, link, !typeid=16
db_table_flair = relation, subreddit, account, !typeid=17
db_table_promocampaign = thing, !typeid=8
```

> Are there any plans to make that the default?

I don't think anyone is actively working on this, so "plans" is a bit strong. That said, if it works out of the box I don't see why we couldn't add this to the example.ini.
